derive_client/_bridge/client.py

```diff
@@ -324,173 +358,164 @@ def deposit_drv(self, amount: int) -> BridgeTxResult:
-        if not target_tx.tx_hash:
-            match tx_result.bridge:
-                case BridgeType.SOCKET:
-                    event_log = self.fetch_socket_event_log(tx_result)
-                case BridgeType.LAYERZERO:
-                    event_log = self.fetch_lz_event_log(tx_result)
-                case _:
-                    raise ValueError()
-            target_tx.tx_hash = event_log["transactionHash"].to_0x_hex()
+        if tx_result.status is not TxStatus.PENDING:
+            raise AlreadyFinalizedError(f"Bridge already in final state: {tx_result.status.name}")
+
+        bridge_event_fetchers = {
+            BridgeType.SOCKET: self.fetch_socket_event_log,
+            BridgeType.LAYERZERO: self.fetch_lz_event_log,
+        }
+        if (fetch_event := bridge_event_fetchers.get(tx_result.bridge)) is None:
+            raise ValueError(f"Invalid bridge_type: {tx_result.bridge}")
+
+        direction = "withdraw" if tx_result.source_chain == ChainID.DERIVE else "deposit"
+        context = self._make_bridge_context(
+            direction=direction,
+            bridge_type=tx_result.bridge,
+            currency=tx_result.currency,
+        )
+
+        # 1. TimeoutError as exception during source_tx.tx_receipt
+        if not tx_result.source_tx.tx_receipt:
+            print(
+                f"⏳ Checking source chain [{tx_result.source_chain.name}] tx receipt for {tx_result.source_tx.tx_hash}"
+            )
+            tx_result.source_tx.exception = None
+            try:
+                tx_result.source_tx.tx_receipt = wait_for_tx_receipt(
+                    w3=context.source_w3, tx_hash=tx_result.source_tx.tx_hash
+                )
+            except Exception as e:
+                tx_result.source_tx.exception = e
+
+        # 2. target_tx is None (i.e. TimeoutError when waiting for event_log on target chain)
+        if not tx_result.target_tx:
+            try:
+                event_log = fetch_event(tx_result, context)
+                tx_result.target_tx = TxResult(event_log["transactionHash"].to_0x_hex())
+            except TimeoutError:
+                return tx_result
 
         # 3. Timeout waiting for target_tx.tx_receipt
-        if not target_tx.tx_receipt:
-            print(f"⏳ Checking target chain [{tx_result.target_chain.name}] tx receipt for {target_tx.tx_hash}")
-            target_tx.tx_receipt = wait_for_tx_receipt(w3=target_w3, tx_hash=target_tx.tx_hash)
+        if not tx_result.target_tx.tx_receipt:
+            print(
+                f"⏳ Checking target chain [{tx_result.target_chain.name}] tx receipt for {tx_result.target_tx.tx_hash}"
+            )
+            tx_result.target_tx.exception = None
+            try:
+                tx_result.target_tx.tx_receipt = wait_for_tx_receipt(
+                    w3=context.target_w3, tx_hash=tx_result.target_tx.tx_hash
+                )
+            except Exception as e:
+                tx_result.target_tx.exception = e
 
```

Comment:
this is identical to the `source_tx.tx_receipt` being absent above (first case). Similarly setting `tx_result.target_tx.exception = None`